backend/src/test/java/com/map/service/UserServiceTest.java

```diff
@@ -0,0 +1,90 @@
+package com.map.service;
+
+import com.map.dto.UserDto;
+import com.map.exception.UserAlreadyExistException;
+import com.map.model.User;
+import com.map.repository.UserRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+@ContextConfiguration(classes = {UserService.class})
+@ExtendWith(SpringExtension.class)
+class UserServiceTest {
+
+    String url = "url";
+    String password = "password";
+    String username = "username";
+    UserRepository userRepositoryConnection = new UserRepository(url, username, password);
+
+    @Autowired
+    private UserService userService;
+
+    @MockBean
+    private UserRepository userRepository;
+
+    @BeforeEach
+    void init() {
+        when(userRepository.getConnection()).thenReturn(userRepositoryConnection.getConnection());
```

Comment:
`when(userRepository.getConnection()).thenReturn(mock(Connection.class));`

This might also work